pkg/config/config.go

```diff
@@ -81,69 +135,174 @@ func Cmd() *cobra.Command {
+		return configFiles, nil
+	}
+
+	if len(*configFileNames) > 0 {
+		for _, configFileName := range *configFileNames {
+			fc, err := os.ReadFile(configFileName)
+			if err != nil {
+				return nil, fmt.Errorf("reading file %q: %w", configFileName, err)
+			}
+			configFiles = append(configFiles, fc)
+		}
+	}
+	return configFiles, nil
 }
 
-// Config rebuilds the YAML file for using Docker Compose or Docker Swarm.
+// Config rebuilds the deployment files
 //
-// A custom template for the YAML file and YAML configs can be provided.
-func Config(dir string, tplFile []byte, configFiles [][]byte) error {
+// A directory containing custom templates for the deployment files can be specified.
+// If it doesn't exist it will be match against the embedded default templates.
+func Config(dir string, tplDirName *string, configFileNames *[]string) error {
+	var err error
+
 	// Create directory
-	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+	if err = os.MkdirAll(dir, os.ModePerm); err != nil {
 		return fmt.Errorf("creating directory at %q: %w", dir, err)
 	}
 
-	// Create YAML file
-	cfg, err := NewYmlConfig(configFiles)
+	// Create YAML config file
+	cfg, err := NewYmlConfig(configFileNames, dir)
 	if err != nil {
 		return fmt.Errorf("creating new YML config object: %w", err)
 	}
 
-	if err := CreateYmlFile(dir, true, tplFile, cfg); err != nil {
-		return fmt.Errorf("creating YAML file at %q: %w", dir, err)
+	// Create the deployment file(s) from template dir
+	if err := CreateDeploymentFilesFromTree(dir, true, tplDirName, cfg); err != nil {
+		return fmt.Errorf("creating deployment files at %q: %w", dir, err)
 	}
 
 	return nil
 }
 
-// CreateYmlFile builds the YAML file at the given directory. Use a truthy value for force
-// to override an existing file.
-func CreateYmlFile(dir string, force bool, tplFile []byte, cfg *YmlConfig) error {
-	if tplFile == nil {
-		tplFile = defaultDockerComposeYml
+// CreateDeploymentFilesFromTree walks through the FS containing templates and calls
+// CreateDeploymentFile to render them one-by-one.
+func CreateDeploymentFilesFromTree(outdir string, force bool, tplDirName *string, cfg *YmlConfig) error {
+	var err error
+	var tplDirFs fs.FS
+	var dirEntries []fs.DirEntry
+	if tplDirFs, err = ReadTplFiles(tplDirName, cfg); err != nil {
+		return fmt.Errorf("reading template dir: %w", err)
 	}
 
-	marshalContentFunc := func(ws int, v interface{}) (string, error) {
-		y, err := yaml.Marshal(v)
+	// Be compatible when fs only contains a single file
+	// -> mock an FS containing that file but rename to config.filename
+	if dirEntries, err = fs.ReadDir(tplDirFs, "."); err != nil {
+		return fmt.Errorf("listing templates directory: %w", err)
+	}
+	if len(dirEntries) == 1 && !dirEntries[0].IsDir() {
+		var content []byte
+		content, err = fs.ReadFile(tplDirFs, dirEntries[0].Name())
 		if err != nil {
-			return "", err
+			return fmt.Errorf("reading single template file %q: %w", dirEntries[0].Name(), err)
```

Comment:
This is probably not good.
The `filename` config parameter should be obsoleted with these changes.
We may introduce a parameter like `outputDirectory` for resulting files after executing templates.